packages/into-inner-derive/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{DeriveInput, Fields};
+
+#[proc_macro_derive(IntoInner)]
+pub fn into_inner_derive(input: TokenStream) -> TokenStream {
+    let ast: DeriveInput = syn::parse(input).unwrap();
+
+    let name = &ast.ident;
+    let ty = match ast.data {
+        syn::Data::Struct(val) => match val.fields {
+            Fields::Unnamed(fields) if fields.unnamed.len() == 1 => {
+                fields.unnamed.first().map(|field| field.ty.clone())
```

Comment:
is the clone necessary?

Reply:
yes, otherwise it's a reference that doesn't live long enough to be returned. I changed it to `to_owned()` to make it clearer though